Feature/offset time

wolfbiter opened the pull request:
@benji6 was working with your `virtual-audio-graph` package (very sweet, thank you) and realized that `offsetTime` is missing from `BufferSource` node params. https://developer.mozilla.org/en-US/docs/Web/API/AudioBufferSourceNode/start

i need this for my app so added to my own fork. it's pretty simple - i think it still needs updates to test, `virtualGraph.update` (eg added to diffing), and readme

wolfbiter commented:
tests pass locally, not sure why the travis CI is blowing up

wolfbiter commented on package.json:
```diff
@@ -34,7 +34,9 @@
   "devDependencies": {
+    "babel-preset-es2015": "^6.22.0",
     "babel-preset-es2015-rollup": "^3.0.0",
+    "babel-register": "^6.22.0",
```
needed to add these deps to build locally

benji6 commented:
@wolfbiter thanks so much for this! Looks good and I have no problem adding some documentation and a test case then publishing a patch. The only thing I'd question is the extra dependencies you added. I cleared my `node_modules` and did a fresh install of the project and had no problem running `npm t` or `npm run build`. Do you definitely get an error without them installed?

CI failing has me stumped too, I think it's something to do with the `concurrently` package running on Travis, but that never seemed to be an issue. I'll push a simple fix to master now

benji6 commented:
@wolfbiter have fixed CI on master, but you'll need to rebase now - sorry!

wolfbiter commented:
1) this branch should now be rebased against your latest master
2) against your latest master, i removed the deps i added, ran `nombom`, and was able to build your project locally. not sure what was happening the first time but works now!
3) to be clear, when i said 'updates to test' i meant that none of the existing tests check exactly that the `offsetTime` actually affects where the audio file starts playing - i dont actually see a way for WebAudioTestAPI to test that.

couple of notes on implementation
1) fyi i see you also have `oscillatorNode` which is a `startAndStopNode`. i presume adding an additional arg to `start(startTime, offsetTime)` wont affect that
2) fyi im actually running my project on electron. though i was able to confirm that `audioNode.start(startTime, offsetTime)` was correctly being called with the named args passed from the virtual audio graph POJO, im no longer using the default `bufferSource` node (instead im implementing a custom script processor by adding `audioContext.createSoundtouchSource`, specifying `soundtouchSource` as node type, and adding `soundtouchSource` to `data.js#startAndStopNodes`)

benji6 commented:
OK, cool, thanks for this. I've done a manual test and it all seems to be working fine. Gonna merge and push a release :guitar:
